File: Problems/Algorithms/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree/CloneTree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode(int x) { val = x; }
 * }
 */

class Solution {
    public final TreeNode getTargetCopy(final TreeNode original, final TreeNode cloned, final TreeNode target) {
        Queue<TreeNode> queue = new LinkedList<>();
        queue.offer(cloned);
        
        int targetVal = target.val;
        while (!queue.isEmpty()) {
            TreeNode curr = queue.poll();
            if (curr.val == targetVal) {
                return curr;
            }
            
            if (curr.left != null) {
                queue.offer(curr.left);
            }
            if (curr.right != null) {
                queue.offer(curr.right);
            }
        }
        
        return null;
    }
}

File: Problems/Algorithms/1379. Find a Corresponding Node of a Binary Tree in a Clone of That Tree/clone_tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, x):
#         self.val = x
#         self.left = None
#         self.right = None

class Solution:
    def __init__(self):
        self.ans = None

    def getTargetCopy(self, original: TreeNode, cloned: TreeNode, target: TreeNode) -> TreeNode:
        def inorder(original: TreeNode, cloned: TreeNode):
            if original is not None:
                inorder(original.left, cloned.left)
                if original == target:
                    self.ans = cloned
                inorder(original.right, cloned.right)

        inorder(original, cloned)
        return self.ans

